app/views/events/_show.html.erb

```diff
@@ -0,0 +1,75 @@
+   <% time_now = DateTime.now.utc %>
+
+<h1 class="mt-5 text-center text-success">Upcoming Events </h1>
+  <table class="table">
+  <thead>
+    <tr class="text-center">
+      <th scope="col">Host Name</th>
+      <th scope="col">Event Title</th>
+      <th scope="col">Event Date</th>
+      <th colspan="3"></th>
+     
+    </tr>
+  </thead>
+<tbody>
+
+  <% @events.each do |event| %>
+    <% if time_now <= event.date %>
+      <tr class="text-center">
+
+      <td class="text-center"><%= event.creator.username %></td>
+      <td class="text-center"><%= link_to event.title, event %></td>
+      <td><%= event.date %></td>
+
+       <% if set_current_user  && set_current_user.id == event.user_id %>
+        <td class="text-center"><%= link_to 'Edit', edit_event_path(event) %></td>
+        <td></td>
+        <% else %>
+        <td></td>
+        <td></td>
+        <% end %>
+        <% if set_current_user && !set_current_user.attended_events.where(id: event.id).any? %>
+
+          <td><%= link_to 'Attend', attendances_path(event_id: event.id), method: :post, class: "btn btn-success" %></td>
+          <td></td>
+          <% else %>
+          <td><button type="button" class="btn btn-light btn-sm disabled">Already attending</button></td>
+
+        <% end %>
```

Comment:
There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it. Kindly move them to a helper.